source/navitiacommon/navitiacommon/models/__init__.py

```diff
@@ -216,22 +216,48 @@ def has_shape(self):
 
     # Note: the private methods below exist to be mocked in the integration tests
 
-    def _get_all_explicitly_authorized_instances(self):
-        return Instance.query_existing().join(Authorization).filter(Authorization.user_id == self.id).all()
+    def _get_all_explicitly_authorized_instances(self, exclude_backend=None):
+        if exclude_backend:
+            return (
+                Instance.query_existing()
+                .join(Authorization)
+                .filter(Authorization.user_id == self.id)
+                .filter(Instance.autocomplete_backend != exclude_backend)
+                .all()
+            )
+        else:
+            return (
+                Instance.query_existing()
+                .join(Authorization)
+                .filter(Authorization.user_id == self.id)
+                .filter(Instance.autocomplete_backend != exclude_backend)
```

Comment:
I don't think you want this line here :thinking: because both branches of the `if/else` are identical...